infrastructure-playbooks/cephadm-adopt.yml

```diff
@@ -249,26 +249,63 @@
       run_once: true
       delegate_to: '{{ groups[mon_group_name][0] }}'
 
-    - name: generate cephadm ssh key
+    - name: check if there is an existing ssh keypair
+      changed_when: false
+      run_once: true
+      delegate_to: '{{ groups[mon_group_name][0] }}'
+      block:
+        - name: public key present
+          stat:
+            path: "{{ cephadm_ssh_pub_key_path }}"
+          register: ssh_pub_key
+
+        - name: private key present
+          stat:
+            path: "{{ cephadm_ssh_priv_key_path }}"
+          register: ssh_priv_key
+
+    - name: fail if either ssh public or private key is missing
+      fail:
+        msg: "One part of the ssh keypair of user {{ cephadm_ssh_user }} is missing"
+      when: not ssh_pub_key.stat.exists or not ssh_priv_key.stat.exists
+
+    - name: generate cephadm ssh key if there is none
+      when:
+        - not ssh_pub_key.stat.exists
+        - not ssh_priv_key.stat.exists
       command: "{{ ceph_cmd }} cephadm generate-key"
       changed_when: false
       run_once: true
       delegate_to: '{{ groups[mon_group_name][0] }}'
 
+    - name: use existing user keypair for remote connections
+      when:
+        - ssh_pub_key.stat.exists
+        - ssh_priv_key.stat.exists
```

Comment:
```suggestion
      when: not false in (ssh_keys.results | map(attribute='stat.exists') | list)
```

maybe consider using an intermediate `set_fact` given that we have to reuse `(ssh_keys.results | map(attribute='stat.exists') | list)` multiple times (better for readability)